File: server/internal/handler/dto/sdi/project.go
package sdi

import "encoding/json"

// CreateProjectSDI is POST .../projects.
type CreateProjectSDI struct {
	Title       string                          `json:"title" minLength:"1" example:"Q3 launch"`
	Description *string                         `json:"description"`
	Icon        *string                         `json:"icon"`
	Status      string                          `json:"status" description:"planned|in_progress|paused|completed|cancelled" example:"planned"`
	Priority    string                          `json:"priority" description:"urgent|high|medium|low|none" example:"none"`
	LeadType    *string                         `json:"lead_type" description:"member|agent"`
	LeadID      *string                         `json:"lead_id"`
	StartDate   *string                         `json:"start_date" example:"2026-09-01"`
	DueDate     *string                         `json:"due_date" example:"2026-09-30"`
	Resources   []CreateProjectResourceEmbedSDI `json:"resources"`
}

// CreateProjectResourceEmbedSDI is an optional resource on project create.
type CreateProjectResourceEmbedSDI struct {
	ResourceType string          `json:"resource_type"`
	ResourceRef  json.RawMessage `json:"resource_ref"`
	Label        *string         `json:"label"`
	Position     *int32          `json:"position"`
}

// PutProjectSDI is PUT .../projects/{projectID}.
type PutProjectSDI struct {
	Title       *string `json:"title"`
	Description *string `json:"description"`
	Icon        *string `json:"icon"`
	Status      *string `json:"status"`
	Priority    *string `json:"priority"`
	LeadType    *string `json:"lead_type"`
	LeadID      *string `json:"lead_id"`
	StartDate   *string `json:"start_date"`
	DueDate     *string `json:"due_date"`
}

// CreateProjectResourceSDI is POST .../projects/{projectID}/resources.
type CreateProjectResourceSDI struct {
	ResourceType string          `json:"resource_type" example:"github_repo"`
	ResourceRef  json.RawMessage `json:"resource_ref"`
	Label        *string         `json:"label"`
	Position     *int32          `json:"position"`
}

// PutProjectResourceSDI is PUT .../projects/{projectID}/resources/{resourceID}.
type PutProjectResourceSDI struct {
	ResourceRef json.RawMessage `json:"resource_ref"`
	Label       *string         `json:"label"`
	Position    *int32          `json:"position"`
}

File: packages/core/types/project.ts
import { z } from "zod";

export type ProjectStatus =
  | "planned"
  | "in_progress"
  | "paused"
  | "completed"
  | "cancelled";

export type ProjectPriority = "urgent" | "high" | "medium" | "low" | "none";

export const ProjectSchema = z.object({
  id: z.string(),
  organization_id: z.string(),
  workspace_id: z.string(),
  title: z.string(),
  description: z.string(),
  icon: z.string().nullable().optional(),
  status: z.string(),
  priority: z.string(),
  lead_type: z.string().nullable().optional(),
  lead_id: z.string().nullable().optional(),
  start_date: z.string().nullable().optional(),
  due_date: z.string().nullable().optional(),
  revision: z.number(),
  task_count: z.number(),
  done_count: z.number(),
  resource_count: z.number(),
  created_at: z.string(),
  updated_at: z.string(),
});
export type Project = z.infer<typeof ProjectSchema>;

export const ProjectResourceSchema = z.object({
  id: z.string(),
  project_id: z.string(),
  workspace_id: z.string(),
  resource_type: z.string(),
  resource_ref: z.unknown(),
  label: z.string().nullable().optional(),
  position: z.number(),
  created_at: z.string(),
  updated_at: z.string(),
});
export type ProjectResource = z.infer<typeof ProjectResourceSchema>;
